### ISSUE TITLE
[BUG] formValues does not update on props change

### ISSUE THREAD

**Issue by manubb** [erikras/redux-form#3793]
### What is the current behavior?
When a component is decorated with `formValues` using a function argument, connected values are not updated when props of the component change.
For example:
```js
formValues({ person } => ({
  name: `${person}.name`,
}))(Component)
```
When `person` changes, the component is still connected with the name of the person it has when mounted.

### What is the expected behavior?
I am expecting formValues to react to props change.

### Other information
The cause of the problem is straightforward when reading the [sources](https://github.com/erikras/redux-form/blob/v7.2.1/src/createFormValues.js): 
 * `valuesMap` is only computed in `constructor` and is not computed again when props change. 
 *  `this.Component` is not updated either and the `render` method uses the initially computed `this.Component`.

I can PR a patch if you feel this is worth to fix.

**Comment by gustavohenke** [comment]
Great, it seems like you already have the fix! Can you send the fix in a PR?

**Comment by manubb** [comment]
The fix still needs a little work. I will PR this evening or next. Thanks for fast reply.

**Comment by erikras** [comment]
Published fix in [`v7.2.2`](https://github.com/erikras/redux-form/releases/tag/v7.2.2).
